src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (target.hidden === false) {
+      target.health -= 50;
+
+      if (target.health === 0) {
+        Animal.alive.splice(Animal.alive.indexOf(target.name), 1);
```

Comment:
```suggestion
        Animal.alive.splice(Animal.alive.indexOf(target), 1);
```
Does your array contain target.names or just target objects?